test/metabase/query_processor/middleware/cache_test.clj

```diff
@@ -348,10 +349,13 @@
             (u/deref-with-timeout called-promise 500)
             (is (= 1 @save-execution-metadata-count))
             (is (= 1 @update-avg-execution-count))
-            (let [avg-execution-time (query/average-execution-time-ms q-hash)]
+            (let [avg-execution-time (query/average-execution-time-ms q-hash)
+                  res (qp/process-query (qp/userland-query query))]
               (is (number? avg-execution-time))
-              ;; rerun query getting cached results
-              (is (:cached (qp/process-query (qp/userland-query query))))
+              (testing "rerun query getting cached results"
+                (is (:cached res))
+                ;; and user will get indication when was query ran last time
+                (is (:updated_at res)))
```

Comment:
Maybe worth testing the type of res as wall?

Reply:
idk, why? there are other tests checking that query processing works :)

Reply:
Sorry, I meant the type of the timestamp. Ideally we'd even check that it's within some tolerance based on the time things took in the test.

Reply:
Ah, checking it now :)